File: backend/migration-script/shop.js
// const mongo = require('mongodb')
// const axios = require('axios')
// const Config = require("./config")


// async function sendToSearch(sShops) {
//     await axios
//     .post(Config.ShopBulkInsertURL, sShops)
//     .then(res => {
//       console.log(`statusCode: ${res.statusCode}`)
//     })
//     .catch(error => {
//       console.error(error.response.data)
//     })
// }

// function CtoSShop(cShop) {
//     return {
//         id : cShop.id,
//         version: cShop.version,
//         slug: cShop.slug,
//         shop_name: cShop.name,
//         shop_image: cShop.image,
//         owner_number: cShop.owner_name,
//         owner_name: cShop.description,
//         contact_number: cShop.contact_number,
//     }
// }

// (async function main() {
//     let counter = 1
//     let data = []
//     while (counter < 50000) {
//         data = [...data, CtoSShop(counter)]
//         console.log(counter)
//         if (counter % 500 == 0) {
//             await sendToSearch(data)
//             console.log(counter, " shops inserted")
//             data = []
//         }
       
//         counter += 1
//     }
//     console.log("finished bulk inserting shops")
//     process.exit(0)
// })()

File: backend/search/shop.go
package search

// Shop defines Shop type
type Shop struct {
	ID             int64  `json:"id" mapstructure:"id"`
	Version        int64  `json:"version" mapstructure:"version"`
	Slug           string `json:"slug" mapstructure:"slug"`
	Approval       int32  `json:"approval" mapstructure:"approval"`
	ContatctNumber string `json:"contact_number" mapstructure:"contact_number"`
	OwnerName      string `json:"owner_name" mapstructure:"owner_name"`
	OwnerNumber    string `json:"owner_number" mapstructure:"owner_number"`
	ShopImage      string `json:"shop_image" mapstructure:"shop_image"`
	ShopName       string `json:"shop_name" mapstructure:"shop_name"`
}
